doc/platform_json_enhancement.md

```diff
@@ -0,0 +1,125 @@
+# Platform capability file enhancement #
+
+### Rev 0.1
+
+## Table of Content
+
+## Revision
+
+ | Rev |     Date    |            Author            | Change Description   |
+ |:---:|:-----------:|:----------------------------:|----------------------|
+ | 0.1 |             | Arun Saravanan Balachandran  | Initial version      |
+
+## Scope
+
+This document provides information on the enhancements for platform capability file `platform.json`.
+
+## Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+|--------------------------|-------------|
+| BMC | Baseband Management Controller |
+| NOS | Network Operating System |
+| PSU | Power Supply Unit |
+
+## Overview
+
+Each networking switch has a set of platform components (e.g: Fans, PSUs, LEDs, etc.) and these components in each platform can have different characteristics (like supported colors for a LED). In a given platform, the components could be controlled by a dedicated platform controller (like BMC) or the NOS running on the CPU is required to control it and in the former the control of the components from the NOS could be limited.
+
+In SONiC the platform components' supported attributes are made available via Platform API, but certain platform specific capabilties for the components are not available for the applications.
+
+This document provides the enhancement for `platform.json` to address the above issue.
+
+## Design
+
+Currently, `platform.json` is used for providing the expected structure of the platform components and interface details for supporting dynamic port breakout.
+
+### Platform capabilities field
+
+A new set of `capabilities` fields are introduced in platform.json, for providing platform specific capablities on control and characteristics of the components.
+
+For each component's attribute, the defined `capabilities` fields are as follows:
+
+- "control" : A boolean, 'true' if the given attribute can be controlled from the NOS, 'false' otherwise.
+- Attribute specific fields:
+    - status led - "color" - A list of the supported colors.
+    - speed
+        - "minimum" - Minimum recommended fan speed that can be set.
```

Comment:
Could you please elaborate on who will be the consumer of this info? Also as platform facts for the test cases?

Reply:
Currently, these fields would be used as platform facts for platform API test suite.
But, these can also be consulted by any platform daemon before executing the respective APIs.

Reply:
@zzhiyuan: I'm not sure I understand your comment. The main usage of these values is for testing purposes to understand boundaries when testing the device. Can you please elaborate?

Reply:
Ah, I meant more for the latter part, "these can also be consulted by any platform daemon before executing the respective APIs."

Since there is already the fantray abstraction, I believe it is better for platform vendor to control the min/max fan speed which may differ by model, via thermal manager. Having values in platform.json for purpose of testing is fine.

Reply:
Ah. Yes, I agree with you. Platform daemons really shouldn't be referring to these values.